zebra/interface.c

```diff
@@ -1058,6 +1058,8 @@ void if_down(struct interface *ifp)
 
 	/* Delete all neighbor addresses learnt through IPv6 RA */
 	if_down_del_nbr_connected(ifp);
+
+	rib_update_handle_vrf_all(RIB_UPDATE_INTERFACE_DOWN, ZEBRA_ROUTE_KERNEL);
```

Comment:
so this is going to do this processing synchronously. that "handle_vrf_all()" is currently an async event callback, via "rib_update()": why not use that path to do the work? was there some issue with doing it async?

Reply:
that whole system area is a mess of code.  I was just hooking into it quickly.  I need to rethink that whole mess if we want to do it async.  Although I am not sure it matters at this point in time.  Can I add it to the list of things that needs to be reworked and rethought?